.github/guides/SMOOTHNESS.md

```diff
@@ -0,0 +1,75 @@
+
+#### 4. Server Underrun
+
+I've been lying to you a little bit so far. We don't actually need the server and client tick rate to sync up perfectly. What we actually need is the time between map sends to sync up with the client's framerate (IE, the time between mapsends needs to be consistent). This is because appearance info is sent by the server as soon as map sending finishes, the time after when verbs run is something closer to "dead" time (verbs don't even show up on byond's cpu tracking tools, although that's for different reasons).
+
+This means that if say, it takes 70% of the tick to get to sending maps on tick 1, 20% on tick 2, and 90% on tick 3, then clients will see that as jitter "forward" (50% of a tick separates 1 and 2) and then a long delay (170% of a tick separates 2 and 3).
+
+There are ways for byond to fix this jumping "foward" (it could just hold the frame for a little while, though this would be weird with afformentioned network problems), but fixing the delay is harder and would in theory require some form of clientside prediction.
+
+This behavior menas we need to try to "pin" the time map sends to roughly the same % of the tick. In some cases we literally do nothing burning cpu time that could otherwise be spent on verbs because we NEED the mapsends to line up.
+
+### Asides
+
+#### Verbs are Stupid
+
+Verbs are not actually measured in world.cpu, because verbs don't really like, have a start and stop time for execution. Verbs can fire anytime after map sending, and before the next tick starts. You might get no verbs until the very last moment, and then get 10 that want to run all at once.
+
+This is because verbs are executed when they arrive at the server. They're literally just requests from the clients to "do something" remember. They're not allowed to run in map sending or while sleeping procs are going, but afterwards it's basically free game for them.
+
+This means in order to keep them from causing server overrun, we need to keep track of when they're trying to run and queue them up if it's too late to finish on time.
+
+> Verbs can also somtimes run AFTER a tick is over, even if no verbs were holding up the tick before. I frankly have no idea why this happens.
+
+#### Types of "failure"
+
+There's two types of ways to fuck up a transition (sound, animate() call, glide, etc). You can start too early, or start too late.
+
+It's quite rare for "undershooting" to be a problem. It only really shows up if the client gets packets too quickly, which would either happen due to server underrun, or because of network hell leading to the client getting a bunch of packets at once.
+
+Overshooting on the other hand is quite common. We have a very hard time avoiding server overrun because well, code can functionally do whatever it wants. The MC does its best to avoid this but if a subsystem/sleeping proc/verb just decides it's not gonna play ball then we're gonna bite it.
```

Comment:
```suggestion
Overshooting on the other hand is quite common. We have a very hard time avoiding server overrun because code can functionally do whatever it wants. The MC does its best to avoid this but if a subsystem/sleeping proc/verb just decides it's not going to cooperate then we're going to have to eat the cost.
```